### PR TITLE
Create telemetry task and make standard headers

### PR DESCRIPTION
**Pull request by AlexandreHiroyuki**

### CODE REVIEW

**Comment by AlexandreHiroyuki** [comment]
Since each module requires procedural execution (order matters), but blocking execution between tasks isn't desirable, I chose to approach the problem with an Active Object design pattern.

A message pump would be undesirable because certain messages require a previous one to be fully executed (like timed engine movements) before it pops out of the queue, blocking other messages that don't need to wait for others to be executed.

File: esp32-controller/src/env-example [hunk 1]
```diff
@@ -17,5 +17,7 @@
 // -------------------- ENVIRONMENTS
 #define WIFI_SSID ("")
 #define WIFI_PASSWORD ("")
+#define WEBSOCKET_URI ("ws://websocket.org")
```
Comment (Amorim33) [inline esp32-controller/src/env-example]: Isn't it better to use an example from localhost?

File: esp32-controller/src/handler/wsEventHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    case WEBSOCKET_EVENT_DATA: {
+      uint16_t message_type =
+          ((uint16_t)data.data_ptr[0] << 8) | data.data_ptr[1];
+      uint16_t parameter[4];
+      for (uint8_t i = 1; i < 5; i++) {
+        parameter[i] =
+            ((uint16_t)data.data_ptr[2 * i] << 8) | data.data_ptr[(2 * i) + 1];
+      }
+
+      uint16_t duration = str_to_number<uint16_t, uint16_t>(parameter, 4);
```
Comment (Amorim33) [inline esp32-controller/src/handler/wsEventHandler.cpp]: Maybe we should add comments to this block, or abstract the arguments parsing to a separate function.
Personally speaking, the amount of loops and type casts make the code a little bit confusing.

File: esp32-controller/src/tasks/taskTelemetry/taskTelemetry.cpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+// Import External Variables //
+extern Ultrasonic ultrasonic;
+extern esp_websocket_client_handle_t ws_client;
+
+void taskTelemetry(void* pvParameters) {
+  for (;;) {
+    char* message = uint_to_str(ultrasonic.read());
+    esp_websocket_client_send_text(ws_client, message, strlen(message),
+                                   portMAX_DELAY);
```
Comment (Amorim33) [inline esp32-controller/src/tasks/taskTelemetry/taskTelemetry.cpp]: How do we identify the type of the message in the back-end?

File: esp32-controller/src/main.h [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+// Internal Includes //
+#include "env.h"
+#include "handler/wsEventHandler.h"
+#include "modules/EngineController/EngineController.hpp"
+#include "tasks/taskTelemetry/taskTelemetry.h"
+#include "utils/typeConversion.h"
+
+// Define Constants //
+#define MESSAGE_TYPE_FORWARD 'f'
+#define MESSAGE_TYPE_BACKWARD 'b'
```
Comment (Amorim33) [inline esp32-controller/src/main.h]: Maybe we should define the outbound message types also. To be able to identify them in the server.

**Review by Amorim33 — APPROVED** [review]
Awesomee pull request

File: esp32-controller/src/env-example (hunk 1 above)
Comment (AlexandreHiroyuki) [reply esp32-controller/src/env-example]: Yes, thanks for noticing

File: esp32-controller/src/handler/wsEventHandler.cpp (hunk 2 above)
Comment (AlexandreHiroyuki) [reply esp32-controller/src/handler/wsEventHandler.cpp]: It is only a temporary code, I would like to test it and discuss it a bit more before committing to the solution.

File: esp32-controller/src/tasks/taskTelemetry/taskTelemetry.cpp (hunk 3 above)
Comment (AlexandreHiroyuki) [reply esp32-controller/src/tasks/taskTelemetry/taskTelemetry.cpp]: An identifier needs to be specified yet, but in the same way I mentioned in the EventHandler comment, I would like to discuss more about how we are doing the communication.

File: esp32-controller/src/main.h (hunk 4 above)
Comment (AlexandreHiroyuki) [reply esp32-controller/src/main.h]: What do you mean? Elaborate, please.

Comment (Amorim33) [reply esp32-controller/src/main.h]:
```suggestion
// Define Constants //

// Inbound //
#define MESSAGE_TYPE_FORWARD 'f'
#define MESSAGE_TYPE_BACKWARD 'b'

// Outbound //
#define MESSAGE_TYPE_TELEMETRY 't'
```

Comment (Amorim33) [reply esp32-controller/src/main.h]: This comment is related to defining the message type for telemetry. 

We can merge it and discuss about it later. this is totally fine. Awesome PR!

**Pull request merged**